pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+
+1. If necessary, creates the directory defined by schema_dir.
+
+2. Downloads the schema from the DMTF web site into the schema_dir.
+
+3. Expands the zip file for the mof into the directory 'mof<schemaVersionNumber'
+   in the schema_dir.
+
+Multiple DMTF schemas may be maintained in the same 'schema_dir' simultaneously
+because each schema is expanded into a directory identified by the
+schema version information.
+
+Once a DMTF schema is expanded into the individual schema files it consumes
+considerable space since it expands to almost 3000 files.  Therefore it is
+logical to remove all of the individual mof files when not being used and
+also if the schema information is to be committed to be used in testing in
+an environment like github. Therefore, a method :meth:`~pywbem.DMTFSchema.clean`
+removes all of the mof files from local storage.  They are restored the next
+time the :class:`DMTFSchema` object for the same version is constructed.
+
+The DMTF maintains two versions of each released schema:
+
+* Final - This schema does not contain any of the schema classes that are
+  marked experimantel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the 'Final' zip file plus other
+  classes that are considered experimental. Typically it is considered less
+  stable.
+
+:class:`DMTFSchema` will optionally load the experimental schema if the
+constructor parameter 'user-experimental' is 'True'.
+
+The DMTFSchema class also provides a method to create a MOF include file
+from the DMTF Schema downloaded that includes only a subset of the classes
+actually in the repository for compile into a MOF repository.
+"""
+import os
+from zipfile import ZipFile
+import shutil
+import six
+
+if six.PY2:
+    # pylint: disable=wrong-import-order
+    from urllib2 import urlopen
+else:
+    # pylint: disable=wrong-import-order
+    from urllib.request import urlopen
+
+
+__all__ = ['DMTFSchema']
+
+
+class DMTFSchema(object):
+    """
+    This class manages the download and install of these files into a form
+    that can be used by pywbem_mock to create a class and qualifier declaration
+    repository.
+
+    It also provides tools to build repositories with DMTF schema.
+    """
+    def __init__(self, schema_ver, schema_dir, use_experimental=False,
+                 verbose=False):
+        """
+        The constructor downloads the schema if the 'schema_dir' does not exist
+        and installs it into the directory defined by the 'schema_dir'
+        attribute.
+
+        If the schema is already installed it simply sets the property values so
+        that the schema can be used.
+
+        If the schema was previously downloaded but not expanded, it is
+        automatically expanded (i.e. the schema/mof subdirectory is created
+        and the mof files expanded into this directory.). This allows only
+        the single file 'schema_mof_filename' to be persisted and the expansion
```

Comment:
There is no file 'schema_mof_filename' and also no parameter or property of that name. I suggest to simply say the zip file.

Reply:
Cleaned up. I removed the sentence 'This allows only the ...' as overkill
DONE